conda/plugins/manager.py

```diff
@@ -657,6 +664,84 @@ def get_environment_specifier(
         else:
             return self.get_environment_specifier_by_name(source=source, name=name)
 
+    def get_environment_exporters(self) -> dict[str, CondaEnvironmentExporter]:
+        """
+        Returns a mapping from export format name to environment exporter.
+        """
+        return {
+            exporter.handler().format: exporter
+            for exporter in self.get_hook_results("environment_exporters")
+            if exporter.handler().format
+        }
+
+    def get_environment_exporter_by_format(
+        self, format_name: str
+    ) -> CondaEnvironmentExporter | None:
+        """
+        Get an environment exporter that supports the given format.
+
+        :param format_name: Format name to find exporter for (e.g., 'json', 'yaml')
+        :return: Environment exporter config or None if not found
+        """
+        for exporter_config in self.get_hook_results("environment_exporters"):
+            exporter = exporter_config.handler()
+            if exporter.format == format_name:
+                return exporter_config
+        return None
+
+    def detect_environment_exporter(
+        self, filename: str
+    ) -> CondaEnvironmentExporter | None:
+        """
+        Detect an environment exporter based on the filename extension.
+
+        :param filename: Filename to find an exporter for (extension is used for detection)
+        :return: CondaEnvironmentExporter that supports the file extension, or None if none found
+        :raises PluginError: If multiple exporters claim to support the same file extension
+        """
+        matches = []
+        for exporter_config in self.get_hook_results("environment_exporters"):
+            exporter = exporter_config.handler()
+            if exporter.can_handle(filename=filename):
+                matches.append(exporter_config)
+
+        if len(matches) == 1:
+            return matches[0]
+        elif len(matches) > 1:
```

Comment:
```suggestion
        elif len(matches) == 0:
            raise PluginError(
                dals(
                     """
                     No environment exporters detected for filename '{filename}'.
                     Choose one exporter explicitly from this list:
                     
                     {", ".join([exporter.name for exporter in self.get_environment_exporters])}
                     """
                )
            )
        elif len(matches) > 1:
```

As per https://github.com/conda/conda/pull/14886#discussion_r2171718782